packages/extension/src/newtab/DndContext.tsx

```diff
@@ -0,0 +1,38 @@
+import usePersistentContext from '@dailydotdev/shared/src/hooks/usePersistentContext';
+import React, { ReactElement, ReactNode } from 'react';
+
+export type DndSettings = { expiration: Date; link: string };
+
+export interface DndContextData {
+  dndSettings: DndSettings;
+  isActive: boolean;
+  onDndSettings: (settings: DndSettings) => Promise<void>;
```

Comment:
Why did you decide to rename it to `on` instead of `set`?

Reply:
Just wanted to make things are uniform and easier to understand. We usually use the prefix `set` for state-related inside the component itself, to easily know that it is coming from either props or a destructured object.